packages/perseus/src/widgets/table/validate-table.ts

```diff
@@ -0,0 +1,26 @@
+import {filterNonEmpty} from "./utils";
+
+import type {PerseusScore} from "../../types";
+import type {PerseusTableUserInput} from "../../validation.types";
+
+function validateTable(userInput: PerseusTableUserInput): PerseusScore {
+    const supplied = filterNonEmpty(userInput);
+
+    const hasEmptyCell = supplied.some(function (row) {
+        return row.some(function (cell) {
+            return cell === "";
+        });
+    });
+
+    if (hasEmptyCell || !supplied.length) {
+        return {
+            type: "invalid",
+            message: null,
+        };
+    }
+
+    // No points earned, but at least we're not invalid
+    return {type: "points", earned: 0, total: 0};
```

Comment:
No action needed: this feels wrong, but I don't know what the alternative is. Maybe return `PerseusScore | null`?

Reply:
See comment above. I'm trying returning `Extract<PerseusScore, {type: "invalid"}> | null`.